### PR TITLE
Add unit test coverage for db/repos/accounts - Closes #1537

### LINKED ISSUE #1537: Add unit test coverage for db/repos/accounts
Parent: #1484

### Expected behavior
We need 100% coverage for this repo

### Actual behavior
Currently its 70% coverage

### PR DESCRIPTION
**Pull request by nazarhussain**
### What was the problem?
Short of unit test coverage for `db/repos/accounts`

### How did I fix it?
Added test cases thoroughly for every method in `db/repos/accounts`. And got 100% coverage for that file. 

### How to test it?
```
npm mocha test/unit/db/repos/accounts.js
```
### Review checklist

* The PR solves #1537 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: test/unit/db/repos/accounts.js [hunk 1]
```diff
@@ -48,871 +50,1404 @@ describe('db', () => {
...
+
+	afterEach(done => {
+		sinonSandbox.restore();
+		seeder
+			.reset(db)
+			.then(() => done(null))
+			.catch(done);
+	});
+
+	it('should initialize db db.accounts repo', () => {
```
Comment (karmacoma) [inline test/unit/db/repos/accounts.js]: `should initialize db.accounts`

File: test/unit/db/repos/accounts.js [hunk 2]
```diff
@@ -48,871 +50,1404 @@ describe('db', () => {
...
+		expect(db.accounts).to.not.null;
```
Comment (karmacoma) [inline test/unit/db/repos/accounts.js]:
```js
expect(db.accounts).to.be.not.null;
```

File: test/unit/db/repos/accounts.js [hunk 3]
```diff
@@ -48,871 +50,1404 @@ describe('db', () => {
...
+
+				expect(db.any.firstCall.args[0]).to.eql(
+					accountsSQL.getOrphanedMemAccounts
+				);
+			});
+
+			it('should yield orphan db.accounts if associated blockId does not exist', function*() {
+				const firstAccount = (yield db.accounts.list({}, ['address']))[0];
+
+				// Let some account to be orphan
```
Comment (karmacoma) [inline test/unit/db/repos/accounts.js]: `// Make an orphan account`

File: test/unit/db/repos/accounts.js [hunk 4]
```diff
@@ -48,871 +50,1404 @@ describe('db', () => {
...
+						firstAccount.address
+					}'`
+				);
+
+				const orphanAccounts = yield db.accounts.getOrphanedMemAccounts();
+				const totalAccounts = yield db.one(
+					'SELECT count(*)::int FROM mem_accounts'
+				);
+
+				// There must be some db.accounts to test
```
Comment (karmacoma) [inline test/unit/db/repos/accounts.js]: `// Check there are some accounts to test`

File: test/unit/db/repos/accounts.js [hunk 5]
```diff
@@ -48,871 +50,1404 @@ describe('db', () => {
...
+				expect(totalAccounts.count).to.above(1);
+
+				expect(orphanAccounts).to.lengthOf(1);
+				expect(orphanAccounts[0].address).to.eql(firstAccount.address);
+			});
+
+			it('should yield orphan db.accounts with address, blockId and id attribute', function*() {
+				// Convert some db.accounts to orphan
```
Comment (karmacoma) [inline test/unit/db/repos/accounts.js]: `// Make an account a orphan`

File: test/unit/db/repos/accounts.js [hunk 6]
```diff
@@ -48,871 +50,1404 @@ describe('db', () => {
...
+				sinonSandbox.spy(db, 'any');
+				yield db.accounts.getDelegates();
+
+				expect(db.any.firstCall.args[0]).to.eql(accountsSQL.getDelegates);
+			});
+
+			it('should return db.accounts with isDelegate set to true ', function*() {
+				const delegates = yield db.accounts.getDelegates();
+
+				// There must be some result to validate
```
Comment (karmacoma) [inline test/unit/db/repos/accounts.js]: `// Check there are some accounts to test`

File: test/unit/db/repos/accounts.js [hunk 7]
```diff
@@ -48,871 +50,1404 @@ describe('db', () => {
...
+			});
+
+			it('should throw error if unknown field is provided', function*() {
+				yield expect(db.accounts.decrement('12L', 'unknown', 1000)).to
+					.eventually.rejected;
+			});
+		});
+
+		describe('remove()', () => {
+			it('should remove a persisted account', function*() {
```
Comment (karmacoma) [inline test/unit/db/repos/accounts.js]: `should remove an existing account`

File: test/unit/db/repos/accounts.js [hunk 8]
```diff
@@ -48,871 +50,1404 @@ describe('db', () => {
...
+				});
 
-					return db.accounts.upsert(originalAccount, 'address').then(() => {
-						return db.accounts.upsert(updatedAccount, 'address').then(() => {
-							return db.accounts
-								.list({ address: originalAccount.address })
-								.then(result => {
-									expect(result.length).to.eql(1);
+				describe('extraCondition', () => {
+					it('should user additional SQL condition if options.extraCondition is provided', function*() {
```
Comment (karmacoma) [inline test/unit/db/repos/accounts.js]: `should use`

File: test/unit/db/repos/accounts.js [hunk 9]
```diff
@@ -48,871 +50,1404 @@ describe('db', () => {
...
-
-				it('should match the multiple conflict keys with AND composite', () => {
-					var originalAccount = accountFixtures.Account();
-					var updatedAccount = accountFixtures.Account({
-						username: originalAccount.username,
-						u_username: originalAccount.u_username,
-					});
 
-					return db.accounts.upsert(originalAccount, 'address').then(() => {
+					it('should return 1 results if options.limit=1', () => {
```
Comment (karmacoma) [inline test/unit/db/repos/accounts.js]: `should return 1 result if options.limit=1`

File: test/unit/db/repos/accounts.js [hunk 10]
```diff
@@ -48,871 +50,1404 @@ describe('db', () => {
...
-					var account = accountFixtures.Account();
-
-					return db.accounts.insert(account).then(result => {
-						expect(result).to.be.null;
-					});
-				});
 
-				it('should insert all columns specified in db.accounts.cs.insert', () => {
-					var account = accountFixtures.Account();
+					it('should skip first 1 result if options.offset=1', () => {
```
Comment (karmacoma) [inline test/unit/db/repos/accounts.js]: `should skip first result if options.offset=1`

File: test/unit/db/repos/accounts.js [hunk 11]
```diff
@@ -48,871 +50,1404 @@ describe('db', () => {
...
 
-					db.accounts
-						.update(null, account)
-						.then(() => {
-							done('should raise error if no address specified');
-						})
-						.catch(reason => {
-							expect(reason.message).to.eql(
-								'Error: db.accounts.update - invalid address argument'
-
```
Comment (karmacoma) [inline test/unit/db/repos/accounts.js]: `should convert a virgin account to a non virgin account`

File: test/unit/db/repos/accounts.js [hunk 12]
```diff
@@ -48,871 +50,1404 @@ describe('db', () => {
...
 						});
+				});
 
-						it('should return "fees" as "bigint"', () => {
-							return db.accounts.list({}, ['fees'], { limit: 1 }).then(data => {
-								expect(data[0].fees).to.be.a('string');
-							});
-						});
+				describe('property names', () => {
+					it('should return "producedBlocks" column if "producedBlocks" is asked', () => {
```
Comment (karmacoma) [inline test/unit/db/repos/accounts.js]: `is requested`

File: test/unit/db/repos/accounts.js [hunk 13]
```diff
@@ -48,871 +50,1404 @@ describe('db', () => {
...
 								});
 						});
+					});
 
-						it('should return "vote" as "bigint"', () => {
-							return db.accounts.list({}, ['vote'], { limit: 1 }).then(data => {
-								expect(data[0].vote).to.be.a('string');
-							});
-						});
+					it('should return "missedBlocks" column if "missedBlocks" is asked', () => {
```
Comment (karmacoma) [inline test/unit/db/repos/accounts.js]: `is requested`

File: test/unit/db/repos/accounts.js (hunk 12 above)
Comment (nazarhussain) [reply test/unit/db/repos/accounts.js]: These tests are removed in PR https://github.com/LiskHQ/lisk/pull/1572 due to renaming the "producedBlocks" column.

File: test/unit/db/repos/accounts.js (hunk 13 above)
Comment (nazarhussain) [reply test/unit/db/repos/accounts.js]: These tests are removed in PR https://github.com/LiskHQ/lisk/pull/1572 due to renaming the "missedBlocks" column.

**Pull request merged**